### PR TITLE
✨Improve msgraph API error handling.

### PR DESCRIPTION
**Pull request by preslavgerchev**
Before:
image

After:
image

### CODE REVIEW

File: resources/packs/azure/azure_ad.go [hunk 1]
```diff
@@ -41,7 +41,7 @@ func (a *mqlAzuread) GetUsers() ([]interface{}, error) {
 	}
 	userList, err := usersClient.Users().Get()
 	if err != nil {
-		return nil, err
+		return nil, msgraphclient.TransformODataError(err)
```
Comment (chris-rock) [inline resources/packs/azure/azure_ad.go]: Should azure really depend on the msgraph package? Is there a odataerror package in the azure sdk

**Review by chris-rock — COMMENTED** [review]
Nice addition

File: resources/packs/azure/azure_ad.go (hunk 1 above)
Comment (preslavgerchev) [reply resources/packs/azure/azure_ad.go]: I can move the error transformation out. The msgraph package has the odataerror in it, we use the `msgraphclient` for users,domains, etc.

This probably brings up a good point of doing the same thing twice: via `ms365` and via `azure` providers. Should we consider dropping one of those?

**Review by chris-rock — APPROVED** [review]
Thank you @preslavgerchev

**Pull request merged**